### PR TITLE
Lookout API add support for Job annotations

### PR DESCRIPTION
**Pull request by carlocamurri**
Adds a new table to Lookout, named `annotation`, to store Job annotations as key/value pairs.

Only annotations that are prefixed with a specified configurable prefix will be added to the table.

Jobs can be looked up by specifying annotation key-value pairs.

Note that the annotations are saved without preserving the prefix. Searches are to be performed without specifying the annotation key prefix.

### CODE REVIEW

File: config/lookout/config.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ metricsPort: 9009
 
 uiConfig:
   armadaApiBaseUrl: "http://localhost:8080"
+  annotationPrefix: "armada/"
```
Comment (jankaspar) [inline config/lookout/config.yaml:7]: lets go for `armadaproject.io/` to follow the domain pattern other things use with annotations.

File: internal/lookout/repository/store.go [hunk 2]
```diff
@@ -257,6 +261,23 @@ func (r *SQLJobStore) upsertContainers(k8sId string, exitCodes map[string]int32)
...
+				"key":    trimmedKey,
+				"value":  value,
+			})
+		}
+	}
+
+	return upsert(r.db, annotationTable, []string{"job_id", "key"}, annotationRecords)
+}
+
 func (r *SQLJobStore) determineJobState() exp.CaseExpression {
```
Comment (robertdavidsmith) [inline internal/lookout/repository/store.go:281]: It's a bit confusing having a data store that filters and trims values. Ideally datastores wouldn't change the data. Could this logic be moved elsewhere? What is the cost of just ditching this logic and storing all annotations including prefix? @jankaspar may have views.

**Review by robertdavidsmith — DISMISSED** [review]
Approving, but suggest @jankaspar design reviews this area once back

File: internal/lookout/repository/store.go (hunk 2 above)
Comment (jankaspar) [reply internal/lookout/repository/store.go:281]: All the annotations are saved in the job table already in json format, this table is intended for lookup by user specified key value pairs. For that saving just prefixed annotations is much easier. We want to for example create a dropdown with all keys in ui. Maybe the name of the table could be improved to reflect this, any suggestions?

Comment (jankaspar) [reply internal/lookout/repository/store.go:281]: Maybe `user_annotations`?

Comment (robertdavidsmith) [reply internal/lookout/repository/store.go:281]: user_annotations?
user_annotations_lookup?
user_annotations_index?

Comment (jankaspar) [reply internal/lookout/repository/store.go:281]: Lets go with `user_annotations_lookup` and rename the methods to match the idea.

**Review by jankaspar — APPROVED** [review]

**Pull request merged**